README.md

```diff
@@ -1,2 +1,115 @@
+uses templates to create each resource with some configuration applied.
+The configuration for a helm chart is called a values file, and is a simple
+YAML document that contains inputs for the templating of the chart.  Each
+environment has its own YAML document containing the required configuration
+for that chart.
+
+But Helm doesn't keep track of the application once it is deployed.  That is,
+it won't notice when the configuration changes and apply those changes.
+ArgoCD (https://argoproj.github.io/argo-cd/) fills this need.  ArgoCD watches
+this repository for new git commits and will keep track of those changes,
+either applying them automatically (sync'ing them), or waiting for an operator
+to press the sync button in the web UI.  ArgoCD is the only layer in this
+stack that has a web UI that can be easily navigated, and it provides many
+useful features, such as deleting resources and resyncing applications.
+
+Now, this is great, because everything is checked into git.  But many service
+configurations require some secrets such as random numbers, certificates, or
+passwords.  But we can't obviously check those into a public repository.  This
+is where Vault (https://www.vaultproject.io/) comes in.  Some of the helm values
+files contain paths to keys in our vault installation, located at
+https://vault.lsst.codes.  Each environment is seeded with a key that it
+uses to access secrets in the vault and make them available for the services
+to read.
+
+## Repository Layout
+
+While ArgoCD can be used and configured in any number of ways, there is also
+a layer of convention to simplify and add some structure that works for us
+to deploy the science platform services.
+
+First, there is the installer directory.  This directory contains a script,
+called install.sh.  The arguments to this are the name of the environment,
+the FQDN, and the read key for https://vault.lsst.codes.  This installer
+script is the entrypoint for setting up a new environment, and should be
+able to be run even on an existing environment to update it.
+
+Next, there is the services directory.  Each sub-directory in services is
+one service of the Rubin Science Platform.  This directory contains helm
+values files for each of the environments that use that component, and
+specify which helm chart to use to deploy that service.  Each of the
+values files are named values-environment.yaml.
+
+Finally, there is the science-platform directory.  This contains an argocd
+application that specifies which services an environment should use, and
+creates all those services in argocd.  The values files in this directory
+contain the service manifest and other top level configuration.
+
+## How to use this repository
+
+Okay now that we understand the tools used, and the various conventions
+("The forms MUST be obeyed..."), we can get into how to use this repository
+to manage environments and services.
+
+There are already a number of different environments, such as int and
```

Comment:
There are already a number of different environments reflecting our many different deployments, such as ...